navigation_onboarding/CMakeLists.txt

```diff
@@ -1,7 +1,33 @@
 cmake_minimum_required(VERSION 3.22.0)
 
-set(CMAKE_CXX_COMPILER clang++-12)
 
-project(nav)
+# -------------------------------------------------- #
+# LIB HELPER                                         #
+# -------------------------------------------------- #
+set(ALL_LIBS "navigation")
+function(make_lib target, include_path)
```

Comment:
I know I did this in the 2022 repo but I think it's bad practice now. Primarily because it encourages you to link everything against everything which isn't really what you want. Maybe have a look at the 2023 repo.

Reply:
On that note, I believe we're going with the following structure.
```
./lib/navigation
./src/my_executable/main.cpp
```
In particular, we never link against anything in `src`